src/sql/workbench/browser/modelComponents/listView.component.ts

```diff
@@ -80,11 +87,16 @@ export default class ListViewComponent extends ComponentBase<azdata.ListViewComp
 	}
 
 	override ngOnDestroy(): void {
+		this.iconClasses.forEach((icon) => {
+			DOM.removeCSSRulesContainingSelector(icon);
+		});
 		this.baseDestroy();
 	}
 
 	public get options(): azdata.ListViewOption[] {
-		return this.getProperties().options ?? [];
+		const options = this.getProperties().options ?? [];
```

Comment:
1. It's odd to only set forceIconPadding when getting options since it shouldn't change unless the properties change. So I'd suggest adding something like this to a setProperties override instead
2. Instead of `forceIconPadding` I'd suggest something more generic like `displayIcon` (and then in the doc comment make clear this will show not only icons but also empty spaces for options without icons)
3. And even better IMO, you could make a CSS rule for showing/hiding the icon fields and then just set the CSS class here on the component itself instead of having the renderer do more calculation